pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/ServerCnx.java

```diff
@@ -270,7 +270,8 @@ protected void handleSubscribe(final CommandSubscribe subscribe) {
                         // creation request either complete or fails.
                         log.warn("[{}][{}][{}] Consumer is already present on the connection", remoteAddress, topicName,
                                 subscriptionName);
-                        ctx.writeAndFlush(Commands.newError(requestId, ServerError.UnknownError,
+                        ServerError error = !existingConsumerFuture.isDone() ? ServerError.ServiceNotReady : ServerError.UnknownError;
+                        ctx.writeAndFlush(Commands.newError(requestId, error,
```

Comment:
I guess @saandrews point is that there is still one case in which broker can return `UnknownError`. Though, after this fix, I think it shouldn't be possible to happen, barring any logical bug